apps/frontend-web/app/create-test/utils/provider.tsx

```diff
@@ -46,30 +34,53 @@ export function CreateTestProvider(props: CreateTestProps) {
       { content: "Our join domains optimize roll we've teeth container." },
       { content: "Hands territories we then later looking buy-in alpha sandwich." },
     ]
-  }]);
-  const [isSaveQuestion, setIsSaveQuestionSate] = useState<boolean>(false);
+  }],
+  addQuestion: (_question: Partial<QuestionItemType>, _isEdit?: boolean, _index?: number,) => { },
+  deleteQuestion: (_index: number) => { }
+}
+
+const CreateContext = createContext<QuestionContext>(initState);
+
+export type QuestionProps = {
+  children: JSX.Element | React.ReactNode;
+};
+
+export function QuestionProvider(props: QuestionProps) {
   const [test, setTestState] = useState<TestItem>(initState.test);
+  const [questions, setQuestionsState] = useState<QuestionItemType[]>([]);
 
-  const saveQuestion = useCallback((newQuestion: QuestionItemType) => {
+  const addQuestion = useCallback((newQuestion: QuestionItemType, isEdit?: boolean, index?: number) => {
```

Comment:
should be
(newQuestion: QuestionItemType, index?: number) =>

if(index >= 0 && index < question.length) =>  newQuestions.splice(index, 1, newQuestion)

else newQuestions.push(newQuestion)